test/test_macros.h

```diff
@@ -56,3 +56,7 @@ bool equals(const T1 &a, const T2 &b, const Tol tolerance) {
  * functions */
 #define ASSERT_NO_THROW_NODISCARD(expr) ASSERT_NO_THROW((void)expr)
 #define ASSERT_THROW_NODISCARD(expr, type) ASSERT_THROW((void)expr, type)
+
+// Helper that casts the result of `statement` to void to suppress warnings.
+#define EXPECT_THROW_DROP_RES(statement, expected_expression)                  \
```

Comment:
```suggestion
#define EXPECT_THROW_DROP_RESULT(statement, expected_expression)                  \
```
?
I find that just `RES` doesn't have a super clear meaning

Reply:
True. I just didn't want it to be too long. But the three extra letter should be fine.